### PR TITLE
Issue #30 Add utility to get client capabilities for LSP server config

### PR DESCRIPTION
**Pull request by mikehaertl**
#30

### CODE REVIEW

File: lua/lsp-file-operations.lua [hunk 1]
```diff
@@ -108,4 +117,19 @@ M.setup = function(opts)
   end
 end
 
+--- The extra client capabilities provided by this plugin. To be merged with
+--- vim.lsp.protocol.make_client_capabilities() and sent to the LSP server.
+M.default_capabilities = function()
```
Comment (mikehaertl) [inline lua/lsp-file-operations.lua:122]: @antosha417 I don't do much lua development - so improvements welcome. If you don't like it we can also remove it again. But that would make setup slightly more verbose.

File: lua/lsp-file-operations.lua [hunk 2]
```diff
@@ -108,4 +117,19 @@ M.setup = function(opts)
...
+  local config = M.config or default_config
+  local result = {
+    workspace = {
+      fileOperations = {}
+    }
+  }
+  for operation, capability in pairs(capabilities) do
+    result.workspace.fileOperations[capability] = config.operations[operation]
+  end
```
Comment (antosha417) [inline lua/lsp-file-operations.lua:131]: We should also add a check here that this capability is enabled in the config. If someone wants to opt out it should be possible.

**Review by antosha417 — CHANGES_REQUESTED** [review]
Looking good! Thank you for your effort!

Added small comment about disabling unwanted capabilities

File: lua/lsp-file-operations.lua (hunk 2 above)
Comment (mikehaertl) [reply lua/lsp-file-operations.lua:131]: That was the idea. Isn't this what's happening? See line 123:

```lua
local config = M.config or default_config
```
So it's using the user config if available. And then `config.operations[operation]` evaluates to `true`/`false` depending on what the user configured.

Comment (antosha417) [reply lua/lsp-file-operations.lua:131]: Oh, yeah, great!

**Review by antosha417 — APPROVED** [review]

**Pull request merged**